node/grpc/server.go

```diff
@@ -149,13 +150,57 @@ func (s *Server) handleStoreChunksRequest(ctx context.Context, in *pb.StoreChunk
 	return &pb.StoreChunksReply{Signature: sigData[:]}, nil
 }
 
+func (s *Server) validateStoreChunkRequest(in *pb.StoreChunksRequest) error {
+	if in.GetBatchHeader() == nil {
+		return api.NewInvalidArgError("missing batch_header in request")
+	}
+	if in.GetBatchHeader().GetBatchRoot() == nil {
+		return api.NewInvalidArgError("missing batch_root in request")
+	}
+	if in.GetBatchHeader().GetReferenceBlockNumber() == 0 {
+		return api.NewInvalidArgError("missing reference_block_number in request")
+	}
+
+	if len(in.GetBlobs()) == 0 {
+		return api.NewInvalidArgError("missing blobs in request")
+	}
+	for _, blob := range in.Blobs {
+		if blob.GetHeader() == nil {
+			return api.NewInvalidArgError("missing blob header in request")
+		}
+		if len(blob.GetHeader().GetQuorumHeaders()) == 0 {
+			return api.NewInvalidArgError("missing quorum headers in request")
+		}
+		if len(blob.GetHeader().GetQuorumHeaders()) != len(blob.GetBundles()) {
+			return api.NewInvalidArgError("the number of quorums must be the same as the number of bundles")
+		}
+		for _, q := range blob.GetHeader().GetQuorumHeaders() {
```

Comment:
Perhaps we should check here that all of the quorumNumbers can fit in a uint8?

Reply:
yeah, I'm thinking a followup to pull out the validation logic of the 2 threshold values, validate retrieval request, as well as address this comment